.gitlab-ci.yml

```diff
@@ -326,13 +326,13 @@ test-linux-stable:                 &test-linux
 unleash-check:
   stage:                           test
   <<:                              *docker-env
-  rules:
-    - if: $CI_PIPELINE_SOURCE == "pipeline"
-      when: never
-    - if: $CI_COMMIT_REF_NAME == "master"
-    - if: $CI_COMMIT_REF_NAME =~ /^v[0-9]+\.[0-9]+.*$/              # i.e. v1.0, v2.1rc1
+  <<:                              *test-refs-no-trigger
   script:
     - cargo install cargo-unleash ${CARGO_UNLEASH_INSTALL_PARAMS}
+    - cargo unleash de-dev-deps
+    # Reuse build artifacts when running checks (cuts down check time by 3x)
+    - mkdir -p target/unleash
+    - export CARGO_TARGET_DIR=target/unleash
```

Comment:
why not put in the `variables:` section above?

Reply:
That's a makeshift optimization that needs addressing in the tool itself and didn't want to "properly" encode it. Should I move it above for consistency for now?

Reply:
I see, it's fine to leave it as is then with the comment it's going to removed once cargo unleash addresses the issue.

Reply:
Added a note